src/tests/test_callback.py

```diff
@@ -1,85 +1,99 @@
-    "testTake2",
-    "xpdAcq_realase",
-)
-
-
-def xpd_filter(x):
-    if "bt_piLast" not in x:
-        return True
-    elif x["bt_piLast"] in xpd_pis:
-        return True
-    else:
-        return False
-
-
-@pytest.mark.parametrize(
-    "idx, dm, bl, f, doc",
-    zip(
-        ["dbes-test-xpd", "dbes-test-iss"],
-        [
-            [("uid", "uid", noconversion), ("bl", "bl", noconversion)],
-            [("uid", "uid", noconversion), ("bl", "bl", noconversion)],
-        ],
-        ["xpd", "iss"],
-        [xpd_filter, lambda x: True],
-        [xpd_doc, iss_doc],
-    ),
-)
-def test_callback(es, idx, dm, bl, f, doc):
-    cb = ElasticInsert(es=es, esindex=idx, docmap=dm, beamline=bl, criteria=f)
-    cb("start", doc)
-    es.indices.flush()
-    res = es.search(idx, body={"query": {"match_all": {}}})
-    assert res["hits"]["total"] == 1
-
-
-def test_no_op_callback(es):
-    cb = ElasticInsert(
-        es=es,
-        esindex="dbes-test-bad_xpd",
-        docmap=[("uid", "uid", noconversion), ("bl", "bl", noconversion)],
-        beamline="xpd",
-        criteria=xpd_filter,
-    )
-    cb("start", xpd_bad_doc)
-    es.indices.flush()
-    res = es.search("dbes-test-bad_xpd", body={"query": {"match_all": {}}})
-    assert res["hits"]["total"] == 0
+from conftest import tdatafile
+from databroker_elasticsearch import callback_from_name
+
+
+cb_config_file = tdatafile('dbes.yml')
+
+
+def indexcount(cb):
+    "Return number of entries in ES index attached to ElasticCallback."
+    es = cb.esindex.es
+    es.indices.refresh()
+    s = es.cat.count(cb.esindex.index, h='count')
+    return int(s)
+
+
+def indexproperties(cb):
+    "Retrieve mapping properties in ES index attached to ElasticCallback."
+    es = cb.esindex.es
+    ei = cb.esindex
+    res = es.indices.get_mapping(ei.index, ignore_unavailable=True)
+    rv = res[ei.index]['mappings'][ei.doc_type]['properties']
+    return rv
+
+
+def require_pi(doc):
+    # only accept measurements by Mingzhao
+    rv = (doc.get('PI', '') == 'Mingzhao')
+    return rv
+
+
+@pytest.fixture
+def cb(es):
+    "Create callback instance from config file, but use our Elasticsearch."
+    cb = callback_from_name(tdatafile('dbes.yml'))
+    cb.esindex.es = es
+    cb.esindex.reset()
+    return cb
+
+
+@pytest.mark.parametrize('criteria,count', [(None, 3), (require_pi, 2)])
+def test_callback_start(cb, criteria, count, issrecords):
+    cb.esindex.criteria = criteria
+    assert indexcount(cb) == 0
+    for doc in issrecords:
+        cb("start", doc)
+    assert indexcount(cb) == count
+    return
+
+
+@pytest.mark.parametrize('criteria,count', [(None, 3), (require_pi, 2)])
+def test_callback_rebuild(cb, criteria, count, issrecords):
+    cb.esindex.criteria = criteria
+    # add 1 dummy document
+    cb.start({"_id": 1, "PI": "Mingzhao"})
+    assert indexcount(cb) == 1
+    # create a mock Header type with "start" attribute
+    Header = collections.namedtuple('Header', 'start')
```

Comment:
Why not use a real header object? That way if the header API moves we'll know.

Reply:
Header is currently not imported at all, so we won't need to do any changes if its API moves. 
The `rebuild` method works with whatever documents sequence as long as they have `.start` attribute.